lib/ABI/FunctionType/Conversion.cpp

```diff
@@ -0,0 +1,480 @@
+/// \file Conversion.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/ADT/SmallSet.h"
+
+#include "revng/ABI/Definition.h"
+#include "revng/ABI/FunctionType/Conversion.h"
+#include "revng/ABI/FunctionType/Support.h"
+#include "revng/ABI/FunctionType/TypeBucket.h"
+#include "revng/Model/Binary.h"
+#include "revng/Model/Helpers.h"
+#include "revng/Support/OverflowSafeInt.h"
+
+static Logger Log("function-type-conversion-to-cabi");
+
+namespace abi::FunctionType {
+
+class ToCABIConverter {
+private:
+  using ArgumentRegisters = SortedVector<model::NamedTypedRegister>;
+  using ReturnValueRegisters = SortedVector<model::TypedRegister>;
+
+public:
+  struct Converted {
+    llvm::SmallVector<model::Argument, 8> RegisterArguments;
+    llvm::SmallVector<model::Argument, 8> StackArguments;
+    model::QualifiedType ReturnValueType;
+  };
+
+public:
+  std::optional<Converted> Result;
```

Comment:
I'd have a `convert` method that contains most of the logic currently in the constructor and returns `Result` instead of having it as a public field.

Reply:
Done.